### PR TITLE
Use global function to compare any entity to both stores

### PR DESCRIPTION
**Pull request by leonorfmartins**
**What is this feature?**

Implement compare for any entity when writing to both stores.
This function is only used for observability purposes.

**Why do we need this feature?**

[Add a description of the problem the feature is trying to solve.]

**Who is this feature for?**

[Add information on what kind of user the feature is for.]

**Which issue(s) does this PR fix?**:

Fixes https://github.com/grafana/search-and-storage-team/issues/44

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: pkg/apiserver/rest/dualwriter.go [hunk 1]
```diff
@@ -207,3 +209,75 @@ func SetDualWritingMode(
...
+	}
+	return h.Sum(nil)
+}
+
+type kv struct {
+	value any
+	key   string
+}
+
+func sortUnstructeredMap(obj map[string]any) []kv {
```
Comment (DanCech) [inline pkg/apiserver/rest/dualwriter.go:242]:
```suggestion
func sortUnstructuredMap(obj map[string]any) []kv {
```

File: pkg/apiserver/rest/dualwriter.go [hunk 2]
```diff
@@ -207,3 +209,75 @@ func SetDualWritingMode(
...
+	cpy := obj.DeepCopyObject()
+	unstObj, err := defaultConverter.ToUnstructured(cpy)
+	if err != nil {
+		return nil
+	}
+	// we don't want to compare meta fields
+	delete(unstObj, "meta")
+
+	ordered := sortUnstructeredMap(unstObj)
+	h := md5.New()
```
Comment (DanCech) [inline pkg/apiserver/rest/dualwriter.go:230]: I'd suggest using https://pkg.go.dev/hash/fnv#New128a , it'll have better performance than md5 and detect changes at least as well.

**Review by DanCech — APPROVED** [review]
Looks great, I made a couple of small suggestions

File: pkg/apiserver/rest/dualwriter.go (hunk 2 above)
Comment (ryantxu) [reply pkg/apiserver/rest/dualwriter.go:230]: given that we have everythign in memory, I don't see any advantage to even hashing...  just copy:
https://github.com/stretchr/testify/blob/v1.9.0/assert/assertions.go#L1760

then we can skip the sorting etc -- writing JSON from map[string] will always be sorted automatically

**Pull request merged**